ecleankernel/process.py

```diff
@@ -130,7 +130,7 @@ def unprefixify(filenames: typing.Iterable[str]
     current = os.uname()[2]
 
     for k in list(remove_kernels):
-        if k.version == current:
+        if k.version.removesuffix(" <UKI> ") == current:
```

Comment:
Really, this is wrong. You're basically asking for trouble if anyplace we forget to strip it. That's why suggested changing the sorting key to a tuple, not injecting a string into version.

Reply:
:O I thought this was what you meant, changing the dictionary key so both entries are different. Which sorting key do we have to change then? I thought the dictionary key was the key that we use for sorting?

Reply:
Update `Kernel` class to keep layout type in a separate var, and update `__eq__`, `__hash__`, `__repr__`.

Feel free to switch it to using `dataclasses` first if you want, that should reduce boilerplate.

Reply:
Okay but this requires touching a bunch of files that the PR does not touch currently. For the moment being I've made this raise an error if we find a kernel version installed in both layouts, as you originally suggested. 
I don't like it, but maybe we can merge the PR like this for now, because otherwise I'm going to end up rebasing and reworking this another dozen times. 
Then I'll try to make the adjustments to the `Kernel` class you suggested in a follow-up PR some time next week.

Reply:
@mgorny is this more like what you had in mind?